config/unix-intel.cmake

```diff
@@ -1,73 +1,77 @@
-  if( "$ENV{CRAY_CPU_TARGET}" STREQUAL "mic-knl" )
-    string( REPLACE "-fp-model=fast" "-fp-model=precise" CMAKE_C_FLAGS_RELEASE
-      ${CMAKE_C_FLAGS_RELEASE} )
+  string(CONCAT CMAKE_C_FLAGS_RELEASE "-O3 -fp-speculation=fast -fp-model=precise -pthread"
+                " -qno-opt-dynamic-align -DNDEBUG")
+  # [KT 2017-01-19] On KNL, -fp-model=fast changes behavior significantly for IMC. Revert to
+  # -fp-model=precise.
+  if("$ENV{CRAY_CPU_TARGET}" STREQUAL "mic-knl")
+    string(REPLACE "-fp-model=fast" "-fp-model=precise" CMAKE_C_FLAGS_RELEASE
+                   ${CMAKE_C_FLAGS_RELEASE})
   endif()
 
-  set( CMAKE_C_FLAGS_MINSIZEREL "${CMAKE_C_FLAGS_RELEASE}" )
-  string( CONCAT CMAKE_C_FLAGS_RELWITHDEBINFO "-g -debug inline-debug-info -O3 -pthread"
-    " -fp-model=precise -fp-speculation=safe -fno-omit-frame-pointer" )
+  set(CMAKE_C_FLAGS_MINSIZEREL "${CMAKE_C_FLAGS_RELEASE}")
+  string(CONCAT CMAKE_C_FLAGS_RELWITHDEBINFO "-g -debug inline-debug-info -O3 -pthread"
+                " -fp-model=precise -fp-speculation=safe -fno-omit-frame-pointer")
 
-  string( APPEND CMAKE_CXX_FLAGS " ${CMAKE_C_FLAGS}")
-  set( CMAKE_CXX_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -early-template-check")
-  set( CMAKE_CXX_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE}")
-  set( CMAKE_CXX_FLAGS_MINSIZEREL "${CMAKE_CXX_FLAGS_RELEASE}")
-  set( CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_C_FLAGS_RELWITHDEBINFO}" )
+  string(APPEND CMAKE_CXX_FLAGS " ${CMAKE_C_FLAGS}")
+  set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -early-template-check")
+  set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE}")
+  set(CMAKE_CXX_FLAGS_MINSIZEREL "${CMAKE_CXX_FLAGS_RELEASE}")
+  set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_C_FLAGS_RELWITHDEBINFO}")
 
-   # Use C99 standard.
-   set( CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99")
+  # Use C99 standard.
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99")
 
 endif()
 
-find_library( INTEL_LIBM m )
-mark_as_advanced( INTEL_LIBM )
+find_library(INTEL_LIBM m)
+mark_as_advanced(INTEL_LIBM)
 
-#--------------------------------------------------------------------------------------------------#
+# -------------------------------------------------------------------------------------------------#
 # Ensure cache values always match current selection
 deduplicate_flags(CMAKE_C_FLAGS)
 deduplicate_flags(CMAKE_CXX_FLAGS)
 force_compiler_flags_to_cache("C;CXX")
 
-# If this is a Cray, the compile wrappers take care of any xHost flags that are needed.
-if( NOT CMAKE_CXX_COMPILER_WRAPPER STREQUAL CrayPrgEnv )
-  set( HAS_XHOST TRUE )
-  toggle_compiler_flag( HAS_XHOST "-xHost" "C;CXX" "")
+# Exceptions for -xHost
+#
+# * If this is a Cray, the compile wrappers take care of any xHost flags that are needed.
+# * On ccs-net we mix & match cpu types (vendor libs are built for sandybridge) and -xHost causes an
+#   ICE when linking libraries.
+if(NOT CMAKE_CXX_COMPILER_WRAPPER STREQUAL CrayPrgEnv AND NOT IS_DIRECTORY "/ccs/opt/bin")
```

Comment:
This is the only 'real' change to `unix-intel.cmake`.  We can't use `-xHost` on ccs-net because we mix and match cpu types.